src/stream/mod.rs

```diff
@@ -222,36 +201,15 @@ impl Stream {
     /// This will download the video to the provided file path.
     #[inline]
     pub async fn download_to<P: AsRef<Path>>(&self, path: P) -> Result<()> {
-        self.internal_download_to(path, None).await
-    }
-
-    /// Attempts to downloads the [`Stream`]s resource.
-    /// This will download the video to the provided file path.
-    /// Takes an [`Callback`](crate::stream::callback::Callback)
-    #[cfg(any(feature = "callback", doc))]
-    #[doc(cfg(feature = "callback"))]
-    #[inline]
-    pub async fn download_to_callback<P: AsRef<Path>>(&self, path: P, callback: Callback) -> Result<()> {
-        self.internal_download_to(path, Some(callback)).await
+        self.internal_download_to(path, None).await.map(|_| ())
     }
 
     #[allow(unused_mut)]
-    async fn internal_download_to<P: AsRef<Path>>(&self, path: P, mut callback: Option<Callback>) -> Result<()> {
+    async fn internal_download_to<P: AsRef<Path>>(&self, path: P, channel: Option<InternalCommunication>) -> Result<PathBuf> {
```

Comment:
I'm not sure if it's possible, but it would be great to remove the need to return a `PathBuf`, since it's ignored most of the times, but requires to turn a `Path` into a `PathBuf`.

Reply:
I need to have a predifined return type for [wrap_callback](https://github.com/fabi321/rustube/blob/fe8f35c75ba93257bdb5e5d783820e721fb571af/src/stream/callback.rs#L268) since the `on_complete_closure` has a `Option<PathBuf>` as argument, and `wrap_callback` thus needs to know the path. Both `download_callback` and `download_to_dir_callback` return a `PathBuf` and that's why I changed this methods return type.

Feel free to share ideas about other solutions.

Reply:
Seems reasonable, and I can't think of a quick way to remove it.